File: php/features.php
<?php
declare(strict_types=1);

// Shikimori SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ShikimoriFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ShikimoriBaseFeature();
            case "test":
                return new ShikimoriTestFeature();
            default:
                return new ShikimoriBaseFeature();
        }
    }
}

File: py/features.py
# Shikimori SDK feature factory

from feature.base_feature import ShikimoriBaseFeature
from feature.test_feature import ShikimoriTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ShikimoriBaseFeature(),
        "test": lambda: ShikimoriTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
